### PR TITLE
Migrate the RegistryClient class from pulp-smash

### PR DESCRIPTION
**Pull request by lubosmj**
Required PR: https://github.com/pulp/pulpcore/pull/3498

[noissue]

### CODE REVIEW

**Comment by lubosmj** [comment]
I am un-drafting this PR.

I wanted to also include the changes where I replaced `pulp_cfg.get_base_url()` with `bindings_cfg.host` in `_local_registry` and `test_cross_repository_blob_mount`. However, intermittent errors were raised during `podman manifest list push oci` operations in `test_push_content`. Not sure if it is worth investigating the root cause at the moment.

File: pulp_container/tests/functional/conftest.py [hunk 1]
```diff
@@ -30,10 +32,76 @@
...
+        # Scheme should not be part of image path, if so, remove it.
+        cmd = (self.name, command) + tuple(args)
+        result = subprocess.check_output(cmd).decode()
+        try:
+            # most client responses are JSONable
+            return json.loads(result)
+        except json.JSONDecodeError:
+            return result
+
+    pull = partialmethod(_dispatch_command, "pull")
```
Comment (decko) [inline pulp_container/tests/functional/conftest.py:81]: :pinched_fingers: Bravo!

File: pulp_container/tests/functional/conftest.py [hunk 2]
```diff
@@ -30,10 +32,76 @@
...
+            os.environ.update(old_environ)
+
+    def _get_registry_client(self):
+        if subprocess.run(("which", self.NAME)).returncode == 0:
+            return self.NAME
+
+        raise RuntimeError("The client '{}' does not appear to be installed.".format(self.NAME))
+
+    def _dispatch_command(self, command, *args):
+        # Scheme should not be part of image path, if so, remove it.
```
Comment (ipanova) [inline pulp_container/tests/functional/conftest.py:72]: leftovers

**Comment by ipanova** [comment]
> I am un-drafting this PR.
> 
> I wanted to also include the changes where I replaced `pulp_cfg.get_base_url()` with `bindings_cfg.host` in `_local_registry` and `test_cross_repository_blob_mount`. However, intermittent errors were raised during `podman manifest list push oci` operations in `test_push_content`. Not sure if it is worth investigating the root cause at the moment.

Eventually these changes will be included too so i believe you're just postponing the investigation, which is fine too.

**Comment by lubosmj** [comment]
The `required PR` reference is here just to verify whether the change in pulpcore has an impact on the CI in pulp_container.

File: pulp_container/tests/functional/conftest.py [hunk 3]
```diff
@@ -30,10 +32,75 @@
...
+        except RuntimeError:
+            raise exc(message)
+
+    @contextmanager
+    def set_env(self, **environ):
+        old_environ = os.environ.copy()
+        os.environ.update(environ)
+        try:
+            yield
+        finally:
```
Comment (ipanova) [inline pulp_container/tests/functional/conftest.py:61]: mm except missing?

Comment (lubosmj) [reply pulp_container/tests/functional/conftest.py:61]: Not really. The goal is the clear up the updated environment variables at all costs.

Comment (ipanova) [reply pulp_container/tests/functional/conftest.py:61]: tbh i have never seen try block without except, got any examples?

Comment (lubosmj) [reply pulp_container/tests/functional/conftest.py:61]: https://docs.python.org/3/reference/compound_stmts.html#finally-clause

Comment (lubosmj) [reply pulp_container/tests/functional/conftest.py:61]: https://github.com/pulp/pulpcore/blob/48d0ef772af1b1484799bd86ba29979a0949be52/pulpcore/app/models/storage.py#L56-L59

Comment (lubosmj) [reply pulp_container/tests/functional/conftest.py:61]: @mdellweg, do you see any added value in logging a raised error? Or, do you think it is sufficient to throw the exception to stdout/stderr as it is right now - no `except` statement?

**Review by ipanova — APPROVED** [review]

**Pull request merged**